app/models.py

```diff
@@ -19,27 +19,38 @@ def __init__(self, uid=0, sid=0,subtotal=0.0,products=[]):
         self.subtotal = float(subtotal)
         self.products = products
 
-    def save(self):
-        if self.sid == 0:
-            self.sid = self.__next_index()
+    def save(self,type):
```

Comment:
Bcoz when we save products, we dont need to again increment the index for sid. For this differentiation, I added that.